app/connector/UserDetailsConnector.scala

```diff
@@ -32,20 +30,22 @@ trait UserDetailsConnector {
   def getUserDetails(userDetailsUri: String)(implicit hc: HeaderCarrier) = http.GET[UserDetails](userDetailsUri)
 }
 
-object CredentialRole extends Enumeration {
-  type CredentialRole = Value
-  val User, Unknown = Value
+case class CredentialRole(value: String) extends AnyVal {
```

Comment:
@paoloboni - You have a answered my question. I missed the `AnyVal` inheritance. And also Java String pool will allow only one instance of String in the heap.